linux/service.c

```diff
@@ -486,33 +486,33 @@ srv_main_loop(service_context sc)
     return 0;
 }
 
+
 /*
  * srv_create_context()
  *	create a service context
  */
+static service_context srvctx;
 service_context
 srv_create_context(void)
 {
-    service_context blah;
-
-    if ((blah = (service_context)malloc(sizeof(struct _servcxt))) == NULL) {
+    if ((srvctx = (service_context)malloc(sizeof(struct _servcxt))) == NULL) {
```

Comment:
I missed this when service.c was added but I really don't like hiding pointers behind typedefs, very confusing!  Can we write this instead as:
```
srvctx = malloc(sizeof(*srvctx));
if (!srvctx) {
    /* error path here */
}
```
A cast shouldn't be needed and the sizeof(*thing) will work even if the type of thing changes.